### PR TITLE
Display grid-wrapper as table and insert treegrid right after the original tree.

### PR DESCRIPTION
**Pull request by legendetm**

### CODE REVIEW

**Comment by deitch** [comment]
Hey @legendetm thanks for the PR. The way it currently works is that the `container` is the original tree element, `gridparent` is `container`'s parent. We put the `gridWrapper` (just a `<div>`) as a child of the `gridparent`, and then put a child to the `gridWrapper` as `midWrapper` (just another `<div>`). `midWrapper` gets one `<div>` column for each column of the grid, and the tree column (normally `0`) gets the original element reparented in there.


Kind of go from:

````
gridparent
 |-- container
````

To:

````
gridparent
 |-- gridWrapper
      |-- midWrapper
           |-- col0
                |-- container
                |-- col1 dataCell
                |-- col2 dataCell
                     ....
                |-- coln dataCell
````

What you are proposing, I think, looks like this:

````
gridparent
   |-- container
   |-- gridWrapper
      |-- midWrapper
            |-- col0 (I think?)
            |-- col1 dataCell
            |-- col2 dataCell
                 ....
            |-- coln dataCell
````

My questions are:

1. What do we gain from it?
2. Does it break anything else that maps out in preparing the headers and the grid, especially the headers?

**Comment by legendetm** [comment]
I just changed the location where the gridparent is inserted.

For example, I have the following setup:
```
<div>
  <div />
  <jstree />
  <div />
</div>
```
then current code appends the gridparent to the end of the jstree parent:
```
<div>
  <div />
  <div />
  <gridparent />
</div>
```
and I changed it to be inserted after the jstree, so the end result will be:
```
<div>
  <div />
  <gridparent />
  <div />
</div>
```

**Comment by legendetm** [comment]
I just changed the location where the gridwrapper is inserted.

For example, I have the following setup:
```
<div>
  <div />
  <jstree />
  <div />
</div>
```
then current code appends the gridwrapper to the end of the jstree parent:
```
<div>
  <div />
  <div />
  <gridwrapper />
</div>
```
and I changed it to be inserted after the jstree, so the end result will be:
```
<div>
  <div />
  <gridwrapper />
  <div />
</div>
```

**Comment by legendetm** [comment]
If you look at one of the test cases `odd-named-id-test.html` included in the tests folder, you see the following html:
```html
<div id="jstree1">
  <h4>First Tree</h4>
  <div id="jstree"></div>
  <div>Tree grid is <span id="status">loading</span>.</div>
</div>
```
With the current version, it looks like this:
screen shot 2016-02-25 at 08 34 56
After the changes I made, it looks like this:
screen shot 2016-02-25 at 08 36 02

**Comment by deitch** [comment]
Oh, I get it! The current design implicitly assumes that `container` is the only child of `parent`. If it isn't, strange things result. The change ensures that the placement is consistent?

**Comment by legendetm** [comment]
Yes, the change 74cf871 always inserts the `gridwrapper` right after the `jstree`.

**Comment by deitch** [comment]
This breaks some cases. Try this for example https://github.com/deitch/jstree-grid/blob/master/tests/widths-test.html

**Comment by legendetm** [comment]
Ok, I see that it brakes the test `Fixed Width, All Columns Defined, Columns > Width`, but in the same time it fixes some other tests where the size is fixed, for example `Fixed Width, All Columns Defined, Columns < Width`, `Fixed Width, Columns Percentages`, `Fixed Width, All Columns Defined, One auto Column` and some others. 

In the test file `widths-test.html` you have two values `Expected` and `Actuals` where the latter shows the wrapper width not the actual width.

screen shot 2016-03-01 at 14 38 21
screen shot 2016-03-01 at 14 39 01

--------
But after the updates the tables have the correct width

screen shot 2016-03-01 at 14 44 51
screen shot 2016-03-01 at 14 45 19

**Comment by legendetm** [comment]
I see that some more work needs be done to support tables where columns' width > table width, but I'm not sure which way to go.

The simplest way is to add another `div` wrapper with `overflow-x: scroll`, if the table width is present.

**Comment by deitch** [comment]
My mind is swimming trying to get my head around this. I remember significant effort to get all of the alignments right the first time around (then the second, then the third... this has had a lot of iterations).

The question is, why does this break those cases?

**Comment by legendetm** [comment]
The braking change is `.jstree-grid-wrapper {display: table}`.

The current layout with the css `display` overloading is
```html
<div>
  <tr>
    <td></td>
  </tr>
</div>
```
So, now when the first `div` has fixed width and the `tr`is wider, it automatically adds the scrolling to the div. But, because `tr` doesn't have `table` as parent, its width is not correctly shown. 
See more http://stackoverflow.com/questions/3480060/css-displaytable-row-does-not-expand-when-width-is-set-to-100

The layout after my changes is
```html
<table>
  <tr>
    <td></td>
  </tr>
</table>
```
Now, the table and all the cells have correct with, but the second example `Fixed Width, All Columns Defined, Columns > Width` fails, because the table will sizes to its content.

In conclusion, to fix both problems, the layout should be following
```html
<div>
  <table>
    <tr>
      <td></td>
    </tr>
  </table>
<div>
```

**Comment by deitch** [comment]
These have been incorporated separately; take a look at the code. 

Thanks for the PR!
